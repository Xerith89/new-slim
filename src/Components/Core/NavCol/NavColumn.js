import React, { Component } from 'react'
import NavItem from './NavItem'

// This component renders on the left of the screen and provides navigation links for the app

export default class NavColumn extends Component {


    render() {
        
    //This stores the links for the claims section of the Nav
    const claimsLinks = {
        links: [
            "New Claim",
            "Diary Reviews",
            "Reports",
            "Another Link"
        ]
    }

    const accountsLinks = {
        links: [
            "Direct Debits",
            "Other Accounts Stuff",
            "Another Link",
            "Another Link"
        ]
    }

    const underwritingLinks = {
        links: [
            "New Quote",
            "Renewals",
            "Another Link",
            "Another Link"
        ]
    }

    const agencyLinks = {
        links: [
            "Manage Brokers",
            "Other Accounts Stuff",
            "Another Link",
            "Another Link"
        ]
    }
        return (
            <div>
                <div id="navMenu" data-testid="navcol" className="card text-center ml-2 mt-5 mb-5 mr-5" style={{ width: '18rem'}}>
                    <div className="card-body p-0">
                        <NavItem title="Claims" links={claimsLinks.links} />
                        <NavItem title="Underwriting" links={underwritingLinks.links} />
                        <NavItem title="Accounts" links={accountsLinks.links} />
                        <NavItem title="Agency" links={agencyLinks.links} />
                    </div>
                </div>
            </div>
        )
    }
}
