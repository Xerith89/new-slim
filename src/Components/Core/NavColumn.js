import React, { Component } from 'react'
import NavItem from './NavItem'

// This component renders on the left of the screen and provides navigation links for the app

export default class NavColumn extends Component {


    render() {
        
    //Nav column inline style
    const style = {
        width: "20rem",
    }

    //This stores the links for the claims section of the Nav
    const claimsLinks = {
        links: [
            "New Claim",
            "Diary Reviews"
        ]
    }
        return (
            <div>
                <div id="navMenu" data-testid="navcol" className="card text-center ml-2 mt-5" style={style}>
                    <div className="card-body navbar-dark bg-dark pt-0">
                        <NavItem title="Claims" links={claimsLinks.links} />
                    </div>
                </div>
            </div>
        )
    }
}
