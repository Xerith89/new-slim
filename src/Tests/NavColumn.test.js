import React from 'react';
import NavColumn from '../Components/Core/NavColumn';
import {render, cleanup, fireEvent} from '@testing-library/react';


afterEach(cleanup)

test('NavColumn component renders to the DOM', () => {
    const {getByTestId} = render(<NavColumn />);
    expect(getByTestId('navcol')).toMatchSnapshot();
});