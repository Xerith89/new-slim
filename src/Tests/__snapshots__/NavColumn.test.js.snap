// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavColumn component renders to the DOM 1`] = `
<div
  class="card text-center ml-2 mt-5"
  data-testid="navcol"
  id="navMenu"
  style="width: 20rem;"
>
  <div
    class="card-body navbar-dark bg-dark pt-0"
  >
    <div>
      <ul
        class="list-group list-group-flush"
      >
        <h3
          class="navbar-dark bg-dark text-white list-group-item"
        >
          Claims
        </h3>
        <li
          class="list-group-item"
        >
          New Claim
        </li>
        <li
          class="list-group-item"
        >
          Diary Reviews
        </li>
      </ul>
    </div>
  </div>
</div>
`;
